clang_tidy/clang_tidy.bzl

```diff
@@ -0,0 +1,156 @@
+load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
+load("//cc:defs.bzl", "BINARY", "LIBRARY")
+
+def _flatten(input_list):
+    return [item for sublist in input_list for item in sublist]
+
+def _run_tidy(ctx, wrapper, exe, additional_deps, config, flags, compilation_contexts, infile, discriminator):
+    inputs = depset(direct = [infile, config] + additional_deps.files.to_list() + ([exe.files_to_run.executable] if exe.files_to_run.executable else []), transitive = [compilation_context.headers for compilation_context in compilation_contexts])
+
+    args = ctx.actions.args()
+
+    # specify the output file - twice
+    outfile = ctx.actions.declare_file(
+        "bazel_clang_tidy_" + infile.path + "." + discriminator + ".clang-tidy.yaml",
+    )
+
+    # this is consumed by the wrapper script
+    if len(exe.files.to_list()) == 0:
+        args.add("clang-tidy")
+    else:
+        args.add(exe.files_to_run.executable)
+
+    args.add(config.path)
+
+    args.add(outfile.path)  # this is consumed by the wrapper script
+    args.add("--export-fixes", outfile.path)
+
+    # add source to check
+    args.add(infile.path)
+
+    # start args passed to the compiler
+    args.add("--")
+
+    # add args specified by the toolchain, on the command line and rule copts
+    args.add_all(flags)
+
+    # add defines
+    for define in _flatten([compilation_context.defines.to_list() for compilation_context in compilation_contexts]):
+        args.add("-D" + define)
+
+    for define in _flatten([compilation_context.local_defines.to_list() for compilation_context in compilation_contexts]):
+        args.add("-D" + define)
+
+    # add includes
+    for i in _flatten([compilation_context.framework_includes.to_list() for compilation_context in compilation_contexts]):
+        args.add("-F" + i)
+
+    for i in _flatten([compilation_context.includes.to_list() for compilation_context in compilation_contexts]):
+        if "gflags" in i:
```

Comment:
Workaround for gflag that I described in [the ticket](https://swift-nav.atlassian.net/browse/BUILD-304?focusedCommentId=68693).